src/cgame/default/cg_hud.c

```diff
@@ -136,40 +151,56 @@ static void Cg_DrawVital(r_pixel_t x, const int16_t value, const int16_t icon, i
 		color = HUD_COLOR_STAT_MED;
 	}
 
-	const char *string = va("%3d", value);
+	const char *string = va("%d", value);
 
 	x += cgi.view->viewport.x;
-	cgi.DrawString(x, y, string, color);
 
-	x += cgi.StringWidth(string);
-	y = cgi.view->viewport.y + cgi.view->viewport.h - HUD_PIC_HEIGHT;
+	if (mirror) {
+		x -= HUD_PIC_HEIGHT;
+		y = cgi.view->viewport.y + cgi.view->viewport.h - HUD_PIC_HEIGHT;
 
-	cgi.Color(pulse);
-	Cg_DrawIcon(x, y, 1.0, icon);
-	cgi.Color(NULL);
+		cgi.Color(pulse);
+		Cg_DrawIcon(x, y, 1.0, icon);
+		cgi.Color(NULL);
+
+		x -= cgi.StringWidth(string) + HUD_VITAL_PADDING;
+		y = cgi.view->viewport.y + cgi.view->viewport.h - HUD_PIC_HEIGHT + 4;
+
+		cgi.DrawString(x, y, string, color);
+
+	} else {
+		y = cgi.view->viewport.y + cgi.view->viewport.h - HUD_PIC_HEIGHT;
+
+		cgi.Color(pulse);
+		Cg_DrawIcon(x, y, 1.0, icon);
+		cgi.Color(NULL);
+
+		x += HUD_PIC_HEIGHT + HUD_VITAL_PADDING;
+		y = cgi.view->viewport.y + cgi.view->viewport.h - HUD_PIC_HEIGHT + 4;
+
+		cgi.DrawString(x, y, string, color);
+	}
 }
 
 /**
  * @brief Draws health, ammo and armor numerics and icons.
  */
 static void Cg_DrawVitals(const player_state_t *ps) {
-	r_pixel_t x, cw, x_offset;
+	r_pixel_t x;
 
 	if (!cg_draw_vitals->integer) {
 		return;
 	}
 
-	cgi.BindFont("large", &cw, NULL);
-
-	x_offset = 3 * cw;
+	cgi.BindFont("large", NULL, NULL);
 
 	if (ps->stats[STAT_HEALTH] > 0) {
 		const int16_t health = ps->stats[STAT_HEALTH];
 		const int16_t health_icon = ps->stats[STAT_HEALTH_ICON];
 
-		x = cgi.view->viewport.w * 0.25 - x_offset;
+		x = (cgi.view->viewport.w / 2.0) - HUD_VITAL_PADDING;
 
-		Cg_DrawVital(x, health, health_icon, HUD_HEALTH_MED, HUD_HEALTH_LOW);
+		Cg_DrawVital(x, true, health, health_icon, HUD_HEALTH_MED, HUD_HEALTH_LOW);
 	}
 
 	if (atoi(cgi.ConfigString(CS_GAMEPLAY)) != GAME_INSTAGIB) {
```

Comment:
One thing I've been meaning to do but haven't gotten around to is finding a common place in cgame to store the results of these cgi.ConfigString fetch operations. Seems silly to parse this every single frame multiple times.